Doc/library/os.rst

```diff
@@ -824,7 +824,7 @@ as internal buffering of data.
       It will always copy no bytes and return 0 as if the file was empty
       because of a known Linux kernel issue.
 
-   .. availability:: Linux kernel >= 4.5 or glibc >= 2.27.
+   .. availability:: Linux >= 4.5, glibc >= 2.27.
```

Comment:
Does this mean “Linux >= 2.6.17 **or** glibc >= 2.5”, or “Linux >= 2.6.17 **with** glibc >= 2.5”?
From the `memfd_create` note it looks like it's the latter, so the comma (which means “or” everywhere else) would be wrong.

Reply:
Python needs both the Linux Kernel syscall and the glibc wrapper to implement the feature. So both conditions must apply.

There are multiple options to handle this:
- We could either clarify this in https://docs.python.org/3/library/intro.html#availability and explain that for a combination of OS and libc, both conditions must hold
- or we could extend the code of the directive class
- or add a remark to the body of the availability directive.

Reply:
What about accepting ` with `, which would just be an alias for `,` for this check?

Reply:
The code does not use the return value of `parse_platforms`. Let's ignore the `with` for now. `Linux >= 3.17 with glibc >= 2.27` is parsed as `{"Linux": "3.17 with glibc >= 2.27"}`.